EventsExpress.Core/Services/TokenService.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using EventsExpress.Core.DTOs;
+using EventsExpress.Core.Infrastructure;
+using EventsExpress.Core.IServices;
+using EventsExpress.Db.Entities;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using AutoMapper;
+
+namespace EventsExpress.Core.Services
+{
+    public class TokenService : ITokenService
+    {
+        private readonly IUserService _userService;
+        private readonly IJwtSigningEncodingKey _signingEncodingKey;
+        private readonly IConfiguration _configuration;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IMapper _mapper;
+
+
+        public TokenService(IConfiguration configuration, IJwtSigningEncodingKey jwtSigningEncodingKey, IUserService userService, IHttpContextAccessor httpContextAccessor, IMapper mapper)
+        {
+            _configuration = configuration;
+            _signingEncodingKey = jwtSigningEncodingKey;
+            _userService = userService;
+            _httpContextAccessor = httpContextAccessor;
+            _mapper = mapper;
+        }
+        public string GenerateAccessToken(UserDTO user)
+        {
+            var lifeTime = _configuration.GetValue<int>("JWTOptions:LifeTime");
+            var claims = new[]
+            {
+                new Claim(ClaimTypes.Email, user.Email),
+                new Claim(ClaimTypes.Role, user.Role.Name),
+                new Claim(ClaimTypes.Name, user.Id.ToString()),
+            };
+            var jwtToken = new JwtSecurityToken(
+                claims: claims,
+                expires: DateTime.Now.AddSeconds(lifeTime),
+                signingCredentials: new SigningCredentials(
+                        _signingEncodingKey.GetKey(),
+                        _signingEncodingKey.SigningAlgorithm)
+            );
+            return new JwtSecurityTokenHandler().WriteToken(jwtToken);
+        }
+        public RefreshToken GenerateRefreshToken(string ipAddress)
+        {
+            var randomNumber = new byte[32];
+            using (var rng = RandomNumberGenerator.Create())
+            {
+                rng.GetBytes(randomNumber);
+                return new RefreshToken
+                {
+                    Token = Convert.ToBase64String(randomNumber),
+                    Expires = DateTime.Now.AddDays(7),
+                    Created = DateTime.Now,
+                    CreatedByIp = ipAddress
+                };
+            }
+        }
+        public ClaimsPrincipal GetPrincipalFromExpiredToken(string token)
```

Comment:
Shouldn't this be "GetPrincipalFromJWT" ?

Reply:
renamed